### PR TITLE
Adding TBB scheduler handle

### PR DESCRIPTION
**Pull request by Alexandr-Solovev**
# Description
Cherry-picking the changes related to TBB fix from 2024.5 to 2024.6 branch.

### CODE REVIEW

**Comment by samir-nasibli** [comment]
> # Description
> Cherry-picking the changes related to TBB fix from 2024.5 to 2024.6 branch.

please check once again target branch mentioned on the description.

**Comment by Alexandr-Solovev** [comment]
CI with the latest TBB: https://ecmd.jf.intel.com/commander/link/jobDetails/jobs/ef338d54-c6bb-f125-91c5-a4bf010d0e2e

File: cpp/daal/src/threading/threading.cpp [hunk 1]
```diff
@@ -61,6 +61,31 @@ DAAL_EXPORT void _daal_tbb_task_scheduler_free(void *& globalControl)
     }
 }
 
+DAAL_EXPORT void _daal_tbb_task_scheduler_handle_free(void *& schedulerHandle)
+{
+    // #if defined(TARGET_X86_64)
```
Comment (Vika-F) [inline cpp/daal/src/threading/threading.cpp:66]: Please add some descriptive note why this change is necessary. For example:
```suggestion
    // Note: TBB 13 deletes task_scheduler_handle itself during the destruction of thread context

    // #if defined(TARGET_X86_64)
```

**Review by Vika-F — APPROVED** [review]
Approving with minor changes.
